### PR TITLE
Point to a new artifacts location.

### PR DESCRIPTION
**Pull request by godofredoc**
## Description

This is in preparation to move the flutter artifacts to a more secure location.

## Related Issues

Bug:
  go/fxb/64576

## Tests

N/A

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [X] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by jonahwilliams — APPROVED** [review]
LGTM

FYI @jmagman for ios frameworks change

**Comment by jmagman** [comment]
Are both available right now?  Will the old URLs continue to work?
We this URL when publishing to CocoaPods, so we don't want old versions to move: https://github.com/CocoaPods/Specs/blob/master/Specs/c/0/4/Flutter/1.22.4/Flutter.podspec.json#L15

**Comment by godofredoc** [comment]
> Are both available right now? Will the old URLs continue to work?
> We this URL when publishing to CocoaPods, so we don't want old versions to move: https://github.com/CocoaPods/Specs/blob/master/Specs/c/0/4/Flutter/1.22.4/Flutter.podspec.json#L15

Old links will still be there but new releases will need to point to the new location(this is to minimize downtime when moving the artifacts to a new gcp project). I'm testing a change to update the location release and engine builders are uploading to.

**Comment by godofredoc** [comment]
\cc @pcsosinski FYI

**Comment by jmagman** [comment]
> Old links will still be there but new releases will need to point to the new location(this is to minimize downtime when moving the artifacts to a new gcp project). I'm testing a change to update the location release and engine builders are uploading to.

Thanks for the explanation.

I'm guessing we're going to wait to merge this until the next engine artifacts are available at the new URL?  The current engine build isn't at the moment https://storage.googleapis.com/flutter_infra_release/flutter/f3cc39a5d646935d84a7a02241b920d0981d342a/ios-release/artifacts.zip

**Comment by pcsosinski** [comment]
@christopherfujino will this affect releases?

**Comment by godofredoc** [comment]
> > Old links will still be there but new releases will need to point to the new location(this is to minimize downtime when moving the artifacts to a new gcp project). I'm testing a change to update the location release and engine builders are uploading to.
> 
> Thanks for the explanation.
> 
> I'm guessing we're going to wait to merge this until the next engine artifacts are available at the new URL? The current engine build isn't at the moment https://storage.googleapis.com/flutter_infra_release/flutter/f3cc39a5d646935d84a7a02241b920d0981d342a/ios-release/artifacts.zip

That is correct I'll land https://flutter-review.googlesource.com/c/recipes/+/8842 and then I'll land this one. There may still be some downtime while I copy the current artifacts in the sdk to the new bucket. I'm preparing a PSA with the details.

**Comment by christopherfujino** [comment]
> @christopherfujino will this affect releases?

We will have to ensure the recipe change is isolated to the recipe code that is branched @godofredoc

File: dev/bots/prepare_package.dart [hunk 1]
```diff
@@ -17,12 +17,12 @@ import 'package:process/process.dart';
 
 const String chromiumRepo = 'https://chromium.googlesource.com/external/github.com/flutter/flutter';
 const String githubRepo = 'https://github.com/flutter/flutter.git';
-const String mingitForWindowsUrl = 'https://storage.googleapis.com/flutter_infra/mingit/'
+const String mingitForWindowsUrl = 'https://storage.googleapis.com/flutter_infra_release/mingit/'
     '603511c649b00bbef0a6122a827ac419b656bc19/mingit.zip';
-const String gsBase = 'gs://flutter_infra';
+const String gsBase = 'gs://flutter_infra_release';
```
Comment (christopherfujino) [inline dev/bots/prepare_package.dart:22]: We also have to migrate the release manifests (e.g. `gs://flutter_infra/releases/releases_linux.json` etc) which are the source of truth for all published flutter releases which the website reads.

Comment (godofredoc) [reply dev/bots/prepare_package.dart:22]: Done.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Review by hammyofmha — COMMENTED** [review]
[https://github.com/flutter/flutter.git](https://github.com/flutter/flutter.git)

**Pull request merged**